### PR TITLE
Revert "Revert "Fix UninitializedSaveTest prototypes" (#16169)"

### PR DESCRIPTION
**Pull request by ElectroJr**
I haven't actually managed to figure out how to reproduce the crash, but it might be fixed now?
It would be good if If someone could figure out how to actually reproduce it.

### CODE REVIEW

File: Content.Server/Lightning/LightningSystem.cs [hunk 1]
```diff
@@ -37,28 +37,28 @@ private void OnRemove(EntityUid uid, LightningComponent component, ComponentRemo
...
-
-        if (component.CanArc)
-        {
-            while (beamController.CreatedBeams.Count < component.MaxTotalArcs)
-            {
-                Arc(component, args.OtherFixture.Body.Owner, lightningBeam.VirtualBeamController.Value);
 
-                var spriteState = LightningRandomizer();
+        if (!component.CanArc || beamController.CreatedBeams.Count >= component.MaxTotalArcs)
+            return;
```
Comment (ElectroJr) [inline Content.Server/Lightning/LightningSystem.cs:45]: Previously this was incorrectly
```cs
 if (!component.CanArc && beamController.CreatedBeams.Count < component.MaxTotalArcs)
            return;
```

**Comment by keronshb** [comment]
Hold off on merging, reviewing this

**Comment by keronshb** [comment]
https://user-images.githubusercontent.com/54602815/236640008-0dd6091c-f173-4cbf-a0cd-972a12ffaa09.mp4

Tested, spammed lightning, nothing catastrophic.

**Comment by keronshb** [comment]
This should be fine since lightning isn't going ham anymore.

**Pull request merged**